src/components/Post.jsx

```diff
@@ -9,10 +9,15 @@ import BookmarkBorderOutlinedIcon from "@mui/icons-material/BookmarkBorderOutlin
 import { db } from "../lib/firebase";
 import firebase from "firebase/compat/app";
 
-function Post(prop) {
-  const { username, caption, imageUrl, avatar, postId, user } = prop;
+import {  doc, updateDoc } from "firebase/firestore";
+
+function Post(prop) { 
+  const { username, caption, imageUrl, avatar, postId, user ,likecount} = prop;
```

Comment:
Instead of passing individual props of `post` object, how about passing the `post` object itself as a prop to `Post.jsx`?
i.e
Remove `caption`, `imageUrl`, `postId`, `likescount`
And add `post` prop, which is a reference to the post object itself,
Then you can access `caption`, `imageUrl`, `postId`, `likescount` props from it
i.e
```
const {caption, imageUrl, postId, likescount} = props.post
```

Reply:
done